location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,576 @@
+                  'Whether to redownload the YouTube videos in the included'
+                  ' csv file')
+flags.DEFINE_string('src_dir',
+                    'location/lbs/activity/audioset/dataprocessing/example_src_dir',
+                    'The location of the audioset csv file(s) and the '
+                    'ontology.json file')
+flags.DEFINE_string('dest_dir',
+                    'location/lbs/activity/audioset/dataprocessing/example_dest_dir',
+                    'The location of the csv file and downloaded YouTube '
+                    'videos')
+flags.DEFINE_bool('scaled', True, 'Whether to average the extracted features')
+
+
+def parse_metadata(src_dir, filename):
+    """Parses metadata and labels from the audioset csv.
+
+    Iterates through the csv files from audioset and extracts the metadata:
+    video_id, start_time, end_time, and labels
+
+    Args:
+        src_dir: Path to a directory where the audioset csv file is expected to
+            be found.
+        filename: A csv file without the .csv file extension
+
+    Returns:
+        A tuple with the first element being a list of lists containing a
+        video_id, start_time (in seconds), and end_time (in seconds). This list
+        is generated from the inputted csv. The second element is a dictionary
+        with video_id, list of labels key, value pairs
+        For example:
+
+        ([[video_id1, start_time1, end_time1],
+        [video_id2, start_time2, end_time2],
+        [video_id3, start_time3, end_time3]],
+        {video_id1 : [label1, label3],
+        video_id12: [label2]
+        video_id3 : [label2, label3]})
+    """
+    csv_paths = []
+    csv_path = join(src_dir, filename + '.csv')
+    audio_list = []
+    label_dict = {}
+    audio_set = set()
+    with open(csv_path) as csv_file:
+        csv_rows = csv.reader(csv_file, delimiter=',')
+        for row in csv_rows:
+            if row[0][0] == '#':
+                continue
+            new_row = []
+            for element in row:
+                new_row.append(element.strip())
+            label_list = []
+            count = 0
+            for element in new_row:
+                if count == 0:
+                    video_id = element
+                elif count == 1:
+                    start_time = float(element)
+                elif count == 2:
+                    end_time = float(element)
+                elif count >= 3:
+                    if element[0] == '"':
+                        element = element[1:]
+                    if element[-1] == '"':
+                        element = element[:-1]
+                    label_list.append(element)
+                count += 1
+            if video_id in audio_set:
+                continue
+            audio_list.append([video_id, start_time, end_time])
+            label_dict[video_id] = label_list
+            audio_set.add(video_id)
+    return audio_list, label_dict
+
+
+def label_id_search(src_dir, label):
```

Comment:
Performance wise, maybe build a dictionary when the script first starts and use that dictionary to have O(1) label loading?

Reply:
That's a great idea. Thank you.